src/library/scala/math/BigInt.scala

```diff
@@ -289,38 +424,56 @@ final class BigInt(val bigInteger: BigInteger)
 
   /** Returns the bitwise complement of this BigInt
    */
-  def unary_~ : BigInt = new BigInt(this.bigInteger.not())
+  def unary_~ : BigInt =
+    // it is equal to -(this + 1)
+    if (isValidLong && _long != Long.MaxValue) new BigInt(-(_long + 1)) else new BigInt(this.bigInteger.not())
 
   /** Returns true if and only if the designated bit is set.
    */
-  def testBit (n: Int): Boolean = this.bigInteger.testBit(n)
+  def testBit (n: Int): Boolean =
+    if (isValidLong && n <= 63) (_long & (1L << n)) != 0 else this.bigInteger.testBit(n)
```

Comment:
Can’t we split the n >= 63 case inside isValidLong to handle it without allocating the BigInteger?

Reply:
We could, though it's additional logic due to the sign bit.

Reply:
@denisrosset decision here?

Reply:
@SethTisue Has been changed.